### PR TITLE
virsh/nodedev: fix chain test

### PR DESCRIPTION
**Pull request by smitterl**
1. Fix test name, subtest should be nodedev_dumpxml.chain for better discoverability and set definition.
2. Update test to run against a specific device.
3. Allow for a list of devices to be physically present. The test will configure the first one to establish the test precondition.

### CODE REVIEW

**Comment by smitterl** [comment]
```
JOB ID     : 099aa3b3f61565498caf1982c55c0fe4f526b3c7
JOB LOG    : /var/log/avocado/job-results/job-2024-06-20T07.09-099aa3b/job.log
 (1/2) type_specific.io-github-autotest-libvirt.virsh.nodedev_dumpxml.chain.device_type_dasd: PASS (5.54 s)
 (2/2) type_specific.io-github-autotest-libvirt.virsh.nodedev_dumpxml.chain.device_type_css: PASS (5.55 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/log/avocado/job-results/job-2024-06-20T07.09-099aa3b/results.html
JOB TIME   : 11.39 s
```

File: libvirt/tests/cfg/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.cfg [hunk 1]
```diff
@@ -1,8 +1,9 @@
-- virsh.nodedev_dumpxml_chain:
+- virsh.nodedev_dumpxml.chain:
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.cfg:1]: this change impacts case pattern in our jobs.config. Do we really need change this?

Comment (smitterl) [reply libvirt/tests/cfg/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.cfg:1]: Yes, it's better in the long run. This is only run on s390x so the change in our local jobs configuration is rather small in comparison with the benefit it brings IMO

**Review by chunfuwen — APPROVED** [review]
lgtm

File: libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.py [hunk 2]
```diff
@@ -16,28 +19,37 @@ def run(test, params, env):
...
+                value = xml.xmltreefile.findtext(xpath)
+                value = value if value else ""
+                if not re.search(pattern, value):
+                    test.fail("Unexpected value on xpath '%s':"
+                              " '%s' does not match '%s'" %
+                              (xpath, value, pattern))
+            xml = get_nodedev_dumpxml(xml.xmltreefile.findtext("parent"))
+            validate_nodedev_xml(test, xml)
+    finally:
+        if device and device[paths.HEADER["Use"]] != "yes":
```
Comment (liang-cong-red-hat) [inline libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.py:51]: is it device[paths.HEADER["Use"]] == "yes" ?

**Review by liang-cong-red-hat — CHANGES_REQUESTED** [review]
just one comment, others LGTM

File: libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.py (hunk 2 above)
Comment (smitterl) [reply libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.py:51]: Yes, this corresponds to the previous step of setting the devices online:
https://github.com/autotest/tp-libvirt/blob/a49c45ce4c9ef0ae6c12eaa557388e4e06701939/libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.py#L28C8-L29C66

Comment (smitterl) [reply libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml_chain.py:51]: Yes, you're right. Updated. Thanks!

**Review by liang-cong-red-hat — APPROVED** [review]

**Pull request merged**